swarm-derive/src/lib.rs

```diff
@@ -188,11 +188,23 @@ fn build_struct(ast: &DeriveInput, data_struct: &DataStruct) -> TokenStream {
             .enumerate()
             .map(|(field_n, field)| match field.ident {
                 Some(ref i) => quote! {
-                    #[allow(deprecated)]
-                    self.#i.inject_connection_established(&peer_id, &connection_id, endpoint, Some(&failed_addresses.into()), other_established);},
+                    self.#i.on_swarm_event(#from_swarm::ConnectionEstablished(#connection_established {
```

Comment:
Needs a minor (breaking) version bump in `libp2p-swarm-derive`.

Reply:
thanks. updated!